students/TacSPx/lesson03/mailroom_modified.py

```diff
@@ -0,0 +1,166 @@
+
+            input("\nPress Enter for Thank You letter:\n")
+            print(f'''Dear {donor_name},
+
+Thank you for your donation of '${donor_amount}' to XYZ Nonprofit for children! It really makes a huge impact for the children 
+in our community.
+
+Those three hours after the end of the school day can make a crucial difference in kids’ lives.
+Thanks to you, kids have a safe place to go after school. Instead of going home alone while their families are at work,
+our kids are learning to play sports, create art, and improving their grades at our Homework Help Center.  All while
+forming friendships with peers and relationships with adult mentors and tutors.
+
+Thank you again {donor_name}, for your ongoing support of our kids!
+
+Sincerely,
+
+XYZ Nonprofit Agency Director
+''')
+            input("Press Enter to Continue")
+            break
+
+###############################################
+#   #2 Create a Report
+###############################################
+def report():
+    ''' Report: creates a tabular summary report of donor transactions'''
+    while (True):
+    # title
+        donor_sort = []
+        title = "*******************  XYZ Nonprofit Charity  ********************\n" \
+                        "________________________________________________________________"
+        heading_name = ("| Donor Name:", "| Totals $:", "| #-of-Gifts:", "| Avg Gift $: ", "|")
+        header = f"{heading_name[0].upper():<20s} {heading_name[1].upper():<13s}" \
+                    f" {heading_name[2].upper():<6s} {heading_name[3].upper():<14s}|"
+        print(title)
+        print(header)
+        #define format row
+        row = "| {Name:<18s} | ${Total:<10.2f} | {NumbGifts:^11d} | ${AverageGifts:<10.2f} |".format
+        # revisions made below to add sorting by totals
+        # step #1 set-up to sort
+        for i in donor_info:
+            donor_sort.append([i[0], sum(i[1]), len(i[1]), sum(i[1], 0) / len(i[1])])
+        # print(donor_sort) - print for testing
+        # step #2 iterate and print sorted, lambda works here but itemgetter looks cleaner
+        #for i in sorted(donor_sort, key=lambda donor_sort: donor_sort[1], reverse=True): # reverse descending
+        for i in sorted(donor_sort, key=itemgetter(1), reverse=True): # reverse descending
+            print(row(Name=i[0], Total=i[1], NumbGifts=i[2], AverageGifts=i[3]))
+        print("________________________________________________________________")
+        input("Press Enter to Continue")
+        break
+
+###############################################
+#   #3 Exit the Program
+###############################################
+def exit_program():
+    '''
+    Will exit the program
+    :return:
+    '''
+    while (True):
+        exit_choice = input("Would you like to Exit?\nEnter 'y' or 'n': ")
+        if (exit_choice.lower() == 'y'):
+            print("\nExiting!")
+            sys.exit()
+        elif (exit_choice.lower() == 'n'):
+            main()
```

Comment:
should not have main() calls inside functions, this should be a simple `return` statement here